### PR TITLE
Limit the number of subsequent requests to the Maps API using the same payload, params and options

### PR DESCRIPTION
**Pull request by alonsogarciapablo**
Fixes: https://github.com/CartoDB/cartodb.js/issues/1283.

@viddo I'd like to know your thoughts about this approach... CartoDB.js won't try to instantiate the exact same map more than three times with this change.

cc: @rochoa

### CODE REVIEW

File: src/windshaft/map-base.js [hunk 1]
```diff
@@ -52,28 +59,37 @@ var WindshaftMap = Backbone.Model.extend({
...
-      stat_tag: this.get('statTag')
-    };
-
-    if (this.get('apiKey')) {
-      params.api_key = this.get('apiKey');
+    var request = new Request(payload, params, options);
+    if (this._canPerformRequest(request)) {
+      this._trackRequest(request);
+      this._performRequest(request);
     }
```
Comment (alonsogarciapablo) [inline src/windshaft/map-base.js]: We're not doing anything when the request cannot be performed... Should we raise an error?

**Comment by viddo** [comment]
Aight, I wait with reviewing it then, ok?

**Comment by alonsogarciapablo** [comment]
Yes please. I'll ping you here once I'm done.

**Comment by alonsogarciapablo** [comment]
@viddo ready again! With the latest change, request to instantiate map will only be triggered if something has changed (payload, params, options) and one of responses from the 3 previous requests is different.

**Comment by viddo** [comment]
looks legit, not much to say 👍

**Pull request merged**